harness/determined/launch/deepspeed.py

```diff
@@ -113,7 +103,7 @@ def create_deepspeed_env_file() -> None:
                 f.write(f"{k}={v}\n")
 
 
-def create_run_command(master_address: str, hostfile_path: str) -> List[str]:
+def create_run_command(master_address: str, hostfile_path: str, ds_args: List[str]) -> List[str]:
```

Comment:
I'm somewhat hesitant to let users pass args to the deepspeed launcher since our launcher is fairly specific to the pdsh backend and many of the arguments need to be set by us.  The only ones I can see value in exposing are --include and --exclude in case specific nodes/gpus aren't working correctly.

Reply:
Let's remove the argument for now and we can add if users request it.

Reply:
Removed it now, but watch out... it will be a breaking change to put it back.  Anybody whose script had a `--` arg in it would be broken unless they added an extra `--` right at the beginning of their script.

I could add a check that enforces that if `--` is present in the script that it must also be added at the beginning.  That would be a really weird thing to enforce but later it would let us add DS_OVERRIDES back in without a breaking change.  My judgement call was "that's too weird" so the current PR doesn't do that.